### PR TITLE
[NCL-8118] Implement Build detail page (Build Log, Alignment Log)

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/LogViewer/LogViewer.tsx [hunk 1]
```diff
@@ -105,24 +112,28 @@ export const LogViewer = ({ data }: ILogViewerProps) => {
           <Button
             onClick={() => {
               setIsPaused(false);
+              logViewerRef.current?.scrollToItem(lineCount);
```
Comment (matedo1) [inline src/components/LogViewer/LogViewer.tsx:115]: Why this change is needed?

**Review by matedo1 — CHANGES_REQUESTED** [review]
- there is uncaught runtime error when build is for example canceled or rejected
- show ssh credentials feature is missing
- download feature is missing
- "Wrap Lines" feature does unwanted content modifications (it removes blank characters)

I will continue with the review once mentioned changes are done.

File: src/components/LogViewer/LogViewer.tsx [hunk 2]
```diff
@@ -105,24 +112,28 @@ export const LogViewer = ({ data }: ILogViewerProps) => {
...
-            label="Force Following"
-            labelOff="Force Following"
-            isChecked={isFollowing}
-            onChange={(checked) => {
-              setIsFollowing(checked);
-              window.localStorage.setItem('log-viewer-following', `${checked}`);
-            }}
-          />
-        </ToolbarItem>
+        {!!customActions?.length && customActions.map((node, index) => <ToolbarItem key={index}>{node}</ToolbarItem>)}
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/LogViewer/LogViewer.tsx:123]: I see no reason why this should be a bad thing.

File: src/services/buildApi.ts [hunk 3]
```diff
@@ -55,6 +55,28 @@ export const getBuild = ({ id }: IBuildApiData, requestConfig: AxiosRequestConfi
...
 
+/**
+ * Gets Build Log of a Build.
+ *
+ * @param serviceData - object containing:
+ *  - id - Build ID
+ * @param requestConfig - Axios based request config
+ */
+export const getBuildLog = ({ id }: IBuildApiData, requestConfig: AxiosRequestConfig = {}) => {
+  return pncClient.getHttpClient().get<string>(`/builds/${id}/logs/build`, requestConfig);
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/services/buildApi.ts:66]: If you mean whether it's the `string[]`, then no. The correct type is `string`.

File: src/components/LogViewer/LogViewer.tsx [hunk 4]
```diff
@@ -30,11 +32,14 @@ interface IOnScrollProps {
...
  * ```
  *
+ * @param variant - static variant if whole log is available, live variant for dynamically loaded log lines
  * @param data - data log viewer will render
  */
-export const LogViewer = ({ data }: ILogViewerProps) => {
+export const LogViewer = ({ variant, data, customActions }: ILogViewerProps) => {
   const logViewerRef = useRef<any>();
 
+  const isStaticVariant = useMemo(() => variant === 'static', [variant]);
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/LogViewer/LogViewer.tsx:41]: Good point.

File: src/components/LogViewer/LogViewer.tsx (hunk 1 above)
Comment (patrikk0123) [reply src/components/LogViewer/LogViewer.tsx:115]: Disabling the pause state works only for the dynamic log variant. The static log variant needs 'scrolling to item' .

File: src/services/buildApi.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (matedo1) [reply src/services/buildApi.ts:66]: No, I mean `responseType` property on axios, the correct type will be probably `text`, because default one is `json`

File: src/components/LogViewer/LogViewer.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (matedo1) [reply src/components/LogViewer/LogViewer.tsx:123]: I would keep it group with other related actions

File: src/components/BuildLogLink/BuildLogLink.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import * as webConfigService from 'services/webConfigService';
+
+interface IBuildLogLinkProps {
+  buildId: string;
+  title?: string;
+}
+
+export const BuildLogLink = ({ buildId, title = 'view' }: IBuildLogLinkProps) => (
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/BuildLogLink/BuildLogLink.tsx:8]: There is an issue with the backend when trying to download the logs, I created a Jira: NCL-8265.

[the remark this answers is not in the document]
Comment (matedo1) [reply src/components/BuildLogLink/BuildLogLink.tsx:8]: done

File: src/services/buildApi.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (matedo1) [reply src/services/buildApi.ts]: done

File: src/services/buildApi.ts [hunk 6]
```diff
@@ -55,6 +55,28 @@ export const getBuild = ({ id }: IBuildApiData, requestConfig: AxiosRequestConfi
...
+
+/**
+ * Gets Alignment Log of a Build.
+ *
+ * @param serviceData - object containing:
+ *  - id - Build ID
+ * @param requestConfig - Axios based request config
+ */
+export const getAlignmentLog = ({ id }: IBuildApiData, requestConfig: AxiosRequestConfig = {}) => {
+  return pncClient.getHttpClient().get<string>(`/builds/${id}/logs/align`, requestConfig);
```
[the remark this answers is not in the document]
Comment (matedo1) [reply src/services/buildApi.ts]: done

File: src/components/LogViewer/LogViewer.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (matedo1) [reply src/components/LogViewer/LogViewer.tsx]: done

File: src/components/LogViewer/LogViewer.tsx (hunk 1 above)
Comment (matedo1) [reply src/components/LogViewer/LogViewer.tsx:113]: done
